### PR TITLE
Dapp ratings

### PR DESCRIPTION
**Pull request by maxaleks**
## Description and Related Issue(s)

Resolves https://github.com/blockscout/frontend/issues/2015 and https://github.com/blockscout/frontend/issues/1972

New env vars:
- `NEXT_PUBLIC_MARKETPLACE_RATING_AIRTABLE_API_KEY`
- `NEXT_PUBLIC_MARKETPLACE_RATING_AIRTABLE_BASE_ID`

## Checklist for PR author
- [x] I have tested these changes locally.
- [ ] I added tests to cover any new functionality, following this [guide](./CONTRIBUTING.md#writing--running-tests)
- [ ] Whenever I fix a bug, I include a regression test to ensure that the bug does not reappear silently.
- [x] If I have added, changed, renamed, or removed an environment variable, I have updated the list of environment variables in the [documentation](ENVS.md) and  made the necessary changes to the validator script according to the [guide](./CONTRIBUTING.md#adding-new-env-variable)

### CODE REVIEW

File: ui/marketplace/Rating/useRatings.tsx [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+
+import config from 'configs/app';
+import useApiQuery from 'lib/api/useApiQuery';
+import useToast from 'lib/hooks/useToast';
+import type { EventTypes, EventPayload } from 'lib/mixpanel/index';
+import * as mixpanel from 'lib/mixpanel/index';
+import { ADDRESS_COUNTERS } from 'stubs/address';
+
+const feature = config.features.marketplace;
+const base = (feature.isEnabled && feature.rating) ?
```
Comment (tom2drum) [inline ui/marketplace/Rating/useRatings.tsx:15]: Can you come up with some more descriptive variable name?

File: ui/marketplace/Rating/useRatings.tsx [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+  const [ isSending, setIsSending ] = useState<boolean>(false);
+  const [ canRate, setCanRate ] = useState<boolean | undefined>(undefined);
+
+  const fetchRatings = useCallback(async() => {
+    if (!base) {
+      return;
+    }
+    const data = await base('apps_ratings').select({ fields: [ 'appId', 'rating' ] }).all();
+    const ratings = formatRatings(data);
+    setRatings(ratings);
```
Comment (tom2drum) [inline ui/marketplace/Rating/useRatings.tsx:67]: Should we somehow handle the error here?

File: ui/snippets/navigation/vertical/__screenshots__/NavigationDesktop.pw.tsx_default_with-tooltips-1.png
Comment (tom2drum) [inline ui/snippets/navigation/vertical/__screenshots__/NavigationDesktop.pw.tsx_default_with-tooltips-1.png:1]: Is this expected? Seems like the offset for this component should be changed.

**Review by tom2drum — CHANGES_REQUESTED** [review]

File: ui/shared/forms/FileSnippet.tsx [hunk 3]
```diff
@@ -76,7 +76,7 @@ const FileSnippet = ({ file, className, index, onRemove, isDisabled, error }: Pr
             <Tooltip
               label={ error }
               placement="top"
-              maxW="320px"
+              maxW={{ base: 'calc(100vw - 8px)', lg: '320px' }}
```
Comment (tom2drum) [inline ui/shared/forms/FileSnippet.tsx:79]: If `calc(100vw - 8px)` is the default value, then here and in other places we can just set a value for desktop viewports:

```suggestion
              maxW={{ lg: '320px' }}
```

**Review by tom2drum — APPROVED** [review]

File: ui/shared/forms/FileSnippet.tsx (hunk 3 above)
Comment (maxaleks) [reply ui/shared/forms/FileSnippet.tsx:79]: No, this will overwrite the default value

**Pull request merged**